### DISCUSSION TITLE
libusb issue

### DISCUSSION THREAD

**Discussion by marchbold** [airsdk/Adobe-Runtime-Support#930] [Q&A]
I'm getting the following issue with the AIR 533 beta when attempting to list and install to iOS devices on macOS catalina 

```
$ adt -devices -platform ios
Could not load libusbmuxd.6.dylib: dlopen(libusbmuxd.6.dylib, 10): image not found
Could not find device list entry point in usbmuxd library
```

Would having a version installed locally be causing a conflict here?

**Comment by ajwfrost** [comment]
Having looked at the `idb` utility, which this calls and which relies upon libusbmuxd, I'm more inclined to believe this is another code-signing issue; perhaps it isn't able to load in that dylib file due to the signing requirements. Oddly, idb and iIPA were both built with the same tools and signed with the same parameters, but there are differences which means that - although it's working on Big Sur - we are seeing a problem with idb on Mojave, and I suspect this Catalina issue is linked.

Are you able to go into the SDK under `lib/aot/bin/iOSBin` and back up your `idb` file before doing:
```
codesign --remove-signature idb
```
and then trying again?

**Reply by marchbold** [reply]
Didn't seem to change anything. 

However I noticed if I run `adt -devices -platform ios` or `./idb -devices` in the `lib/aot/bin/iOSBin` directory it works. 

But any other directory it fails with the same error.

**Reply by ajwfrost** [reply]
Thanks, that's a massive clue. This feels a bit crazy as we didn't recompile that binary! but it looks like the internal referencing has gone a bit wrong... we should be able to fix that, and in the meantime an earlier SDK will have the working version in it (I still suspect the 'idb' tool being the only one that's gone wrong, but we can confirm that......)

**Reply by ajwfrost** [reply]
So, reverting to an earlier one and code-signing it again, and also removing the signature from the zlib library that for some reason was invalid...
Are you able to replace the iOSBin folder with the files from the below message, this will hopefully fix it for the pre-11.x macs..
https://transfer.harman.com/message/T4oPobA23QEvkClkY4TA3J

thanks

**Reply by marchbold** [reply]
Okay, that did the trick, it's now working on catalina (10.15.7) for me.
